### PR TITLE
Force codesign to replace existing signatures

### LINKED ISSUE #6265: Nix + macOS codesigning: `-f` flag needed
The error I get without `-f` is:

```
ocaml5.0.0+alpha1-conan-cli> libc++abi: terminating with uncaught exception of type std::runtime_error: file is already signed. pass -f to sign regardless.
```

_Originally posted by @anmonteiro in https://github.com/ocaml/dune/issues/6260#issuecomment-1284226081_

### PR DESCRIPTION
**Pull request by voodoos**
We add `-f` to the list of flags passed to `codesign`. In some cases, the binary already has a signature so the `codesign` tool from Apple prints some error messages on stderr. We filter out these error messages as they are innocuous. In addition, this ensures that the test suite has the same output on macos and Linux.

Fixes #6265

### CODE REVIEW

**Comment by emillon** [comment]
@ulysse I pushed some changed. Can you check that the test suite passes on M1? (the "replacing existing signature" messages should be gone)

**Comment by emillon** [comment]
@anmonteiro I implemented the fix you suggested. can you try it and let me know if it works for you?

File: src/dune_rules/artifact_substitution.ml [hunk 1]
```diff
@@ -52,8 +52,22 @@ type conf =
   }
 
 let mac_codesign_hook ~codesign path =
-  Process.run ~display:!Clflags.display Strict codesign
-    [ "-s"; "-"; Path.to_string path ]
+  Temp.with_temp_file ~dir:Path.root ~prefix:"codesign" ~suffix:"stderr"
```
Comment (emillon) [inline src/dune_rules/artifact_substitution.ml:55]: Not sure it's the right API to call / the right dir to put the tempfile.

**Comment by anmonteiro** [comment]
> @anmonteiro we implemented the fix you suggested. can you try it and let me know if it works for you?

I don’t actually have access to a m1 machine anymore, and won’t have for at least another 2 weeks. If you don’t mind waiting I could try then.

**Comment by emillon** [comment]
No problem. Since I think the issue only affects m1+nix users I think that we can move that to 3.8. Thanks!

**Comment by anmonteiro** [comment]
I've now confirmed this works on arm64 (M2).

**Comment by emillon** [comment]
Closing in favor of #6975.

**Pull request closed without merging**